src/test/functions.rs

```diff
@@ -39,13 +39,20 @@ fn lifetime() {
             }
         ]
 
-        [ /* TODO */ ]
+        expect_test::expect!["()"]
+    )
+}
+
+#[test]
+fn wf() {
+    crate::assert_ok!(
+        [
+            crate Foo {
+                fn foo<ty T>() -> T { trusted }
+                fn bar(fn foo<u32>, fn(u32) -> (), fn(bool)) -> () { trusted }
+            }
+        ]
 
-        expect_test::expect![[r#"
-            judgment `prove_wc_list { goal: {@ wf(&!lt_0 !ty_1)}, assumptions: {}, env: Env { variables: [!lt_0, !ty_1], bias: Soundness }, decls: decls(222, [], [], [], [], [], [], {}, {}) }` failed at the following rule(s):
-              the rule "some" failed at step #0 (src/file.rs:LL:CC) because
-                judgment `prove_wc { goal: @ wf(&!lt_0 !ty_1), assumptions: {}, env: Env { variables: [!lt_0, !ty_1], bias: Soundness }, decls: decls(222, [], [], [], [], [], [], {}, {}) }` failed at the following rule(s):
-                  the rule "parameter well formed" failed at step #0 (src/file.rs:LL:CC) because
-                    judgment had no applicable rules: `prove_wf { goal: &!lt_0 !ty_1, assumptions: {}, env: Env { variables: [!lt_0, !ty_1], bias: Soundness }, decls: decls(222, [], [], [], [], [], [], {}, {}) }`"#]]
+        expect_test::expect!["()"]
```

Comment:
@nikomatsakis this is a test for fn well-formed. I can also add some tests for catching poorly formed fns.